%YAML:1.0
cameras:
    - {T_SC:
         [ 1,            0,            0,        0.417,
           0,           -1,  1.22465e-16,         0.18,
           0, -1.22465e-16,           -1,       -0.095,
           0,            0,            0,            1], # zed2 imu
       image_dimension: [512, 512],
       distortion_coefficients: [0, 0, 0, 0],
       distortion_type: radialtangential,
       focal_length: [256, 256],
       principal_point: [256, 256]}

    - {T_SC:
         [ 1,            0,            0,        0.417,
           0,           -1,  1.22465e-16,       -0.045,
           0, -1.22465e-16,           -1,       -0.095,
           0,            0,            0,            1], #zed2 imu
       image_dimension: [512, 512],
       distortion_coefficients: [0, 0, 0, 0],
       distortion_type: radialtangential,
       focal_length: [256, 256],
       principal_point: [256, 256]}

camera_params:
    camera_rate: 10
    sigma_absolute_translation: 0.0
    sigma_absolute_orientation: 0.0
    sigma_c_relative_translation: 0.0 # below 1.0e-6 ruines numerics...
    sigma_c_relative_orientation: 0.0 # below 1.0e-6 ruines numerics...

imu_params:
    a_max: 176.0 # acceleration saturation [m/s^2]
    g_max: 7.8 # gyro saturation [rad/s]
    #zed2 imu
    sigma_g_c: 2.4959966280437328e-03 # gyro noise density [rad/s/sqrt(Hz)] 0.00016017
    sigma_a_c: 2.5273067978502026e-02 # accelerometer noise density [m/s^2/sqrt(Hz)] 0.00071376
    sigma_bg: 2.7653291083240783e-04 # gyro bias prior [rad/s]
    sigma_ba: 1.4314525446940051e-04 # accelerometer bias prior [m/s^2]
    sigma_gw_c: 2.7653291083240783e-04 # gyro drift noise density [rad/s^s/sqrt(Hz)] 0.00000165
    sigma_aw_c: 1.4314525446940051e-04 # accelerometer drift noise density [m/s^2/sqrt(Hz)] 0.00002874
    #xsens imu
#    sigma_g_c: 0.0016210361526331097 # gyro noise density [rad/s/sqrt(Hz)] 0.00016017
#    sigma_a_c: 0.005617343328024146 # accelerometer noise density [m/s^2/sqrt(Hz)] 0.00071376
#    sigma_bg: 4.2781957524004126e-05 # gyro bias prior [rad/s]
#    sigma_ba: 0.00014537747959689836 # accelerometer bias prior [m/s^2]
#    sigma_gw_c: 4.2781957524004126e-05 # gyro drift noise density [rad/s^s/sqrt(Hz)] 0.00000165
#    sigma_aw_c: 0.00014537747959689836 # accelerometer drift noise density [m/s^2/sqrt(Hz)] 0.00002874
    tau: 3600.0 # reversion time constant, currently not in use [s]
    g: 9.81007 # Earth's acceleration due to gravity [m/s^2]
    a0: [ 0.0, 0.0, 0.0 ] # Accelerometer bias [m/s^2]
    imu_rate: 100
    # tranform Body-Sensor (IMU)
    T_BS:
        [1.0000, 0.0000, 0.0000, 0.0000,
         0.0000, -1.0000, 0.0000, 0.0000,
         0.0000, 0.0000, -1.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 1.0000]
sonar_params:
    T_SSo:
        [-0.0907363, -0.0521933,   0.994506,   0.203479,
         -0.995553,  0.0301427, -0.0892499,  0.0326268,
         -0.0253189,  -0.998182, -0.0546962,  0.0188909,
          0,          0,          0,          1]  # sonar to zed2 imu
#        [-0.117299, -0.0756147,   0.990214,   0.332616,
#         -0.992957,  0.0256546,  -0.115665,   0.048771,
#         -0.0166576,  -0.996807, -0.0780914, -0.0180802,
#          0,          0,          0,          1] # sonar to xsens imu

# Estimator parameters
numKeyframes: 5 # number of keyframes in optimisation window   5
numImuFrames: 3 # number of frames linked by most recent nonlinear IMU error terms  3

# ceres optimization options
ceres_options:
    minIterations: 3   # minimum number of iterations always performed
    maxIterations: 25  # never do more than these, even if not converged  20
    timeLimit: 0.1   # [s] negative values will set the an unlimited time limit 0.035

# detection
detection_options:
    threshold: 10.0      # detection threshold. By default the uniformity radius in pixels
    octaves: 1           # number of octaves for detection. 0 means single-scale at highest resolution
    maxNoKeypoints: 400  # restrict to a maximum of this many keypoints per image (strongest ones)

# delay of images [s]:
imageDelay: 0.0  # in case you are using a custom setup, you will have to calibrate this. 0 for the VISensor.

# display debug images?
displayImages: true  # displays debug video and keyframe matches. May be slow.
publishDebugImages: false  # publish debug image to topic ~debug_image_x

# use direct driver
useDriver: false

# some options for how and what to publish -- optional in ROS-free version
publishing_options:
    publish_rate: 100                  # rate at which odometry updates are published only works properly if imu_rate/publish_rate is an integer!!
    publishLandmarks: true             # select, if you want to publish landmarks at all
    landmarkQualityThreshold: 1.0e-2   # landmark with lower quality will not be published
    maximumLandmarkQuality: 0.05       # landmark with higher quality will be published with the maximum colour intensity
    maxPathLength: 20                  # maximum length of the published path
    publishImuPropagatedState: true    # Should the state that is propagated with IMU messages be published? Or just the optimized ones?
    # provide custom World frame Wc
    T_Wc_W:
        [1.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 1.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 1.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 1.0000]
    trackedBodyFrame: B                # B or S, the frame of reference that will be expressed relative to the selected worldFrame
    velocitiesFrame: Wc                # Wc, B or S,  the frames in which the velocities of the selected trackedBodyFrame will be expressed in

#Sharmin
isSonarUsed: false
isDepthUsed: false
isForwardSonarUsed: false
isRelocalization: true
histogramMethod: "NONE" # NONE, HISTOGRAM, CLAHE
#claheClipLimit: 0.5
#claheTileGridSize: 0.5
resizeFactor: 1  # by default 1.0. set this value if you want to resize the original image by this factor. You DO NOT need to change camera/projection matrix, those will be updated accordingly.